### PR TITLE
[AKS] list available k8s versions, add tab completers, refactor code

### LINKED ISSUE #5461: Is there a way to know which kubernetes versions are supported?
Is there a way to find out which Kubernetes versions are supported by using "az aks create" cli command?  In your documentation, you mentioned 1.7.7 as default and 1.8.2 as well.  Is there a way to find out supported Kubernetes versions?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d75c3a8d-1187-bea1-a7d4-7ebba71c1790
* Version Independent ID: 75b0c352-762e-16ae-5cc6-d807ffc1dc3f
* [Content](https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#feedback)
* [Content Source](https://github.com/Azure/azure-docs-cli-python/blob/live/latest/docs-ref-autogen/aks.yml)
* Service: unspecified

### PR DESCRIPTION
**Pull request by mboersma**
Note that this is a **breaking change** since it renames a command--see the second and third bullet points below. But AKS is still in preview, and since `az aks get-versions` actually returned only upgrades available, I think this change is warranted. Feedback is more than welcome.

cc: @sauryadas @slack 

* Use `azure-mgmt-containerservice` 3.0.1 package with fixed listOrchestrators endpoint (see Azure/azure-rest-api-specs#2327 and Azure/azure-sdk-for-python#1832)
* Rename `az aks get-versions` to `az aks get-upgrades`, because that's what it did
* Reimplement `az aks get-versions` to return Kubernetes versions available for creating a cluster (using the aforementioned listOrchestrators endpoint)
* Add tab-completion for the `--kubernetes-version` argument to `az aks create`
* Add tab-completion for the `--kubernetes-version` argument to `az aks upgrade`
* Rewrite formatters to use [JMESPath](http://jmespath.org/) queries
* Refer to current, preferred Kubernetes versions 1.7.12 and 1.8.7 consistently
* Update help message to refer to correct service principal cache file for AKS
* Add new commands to AKS scenario tests and re-record them all

Closes #5461

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [X] Each command and parameter has a meaningful description.
- [X] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5475
This is an **experimental preview** for @microsoft users.

**Comment by mboersma** [comment]
@brendandburns and the rest of the AKS team thought it was reasonable to make this breaking change at this point, if the CLI team is ok with it. It's unlikely anyone has built scripting around the `az aks get-versions` command yet, and this will be more coherent going forward.

**Comment by sauryadas** [comment]
I think its okay to break while we are in preview. Unfortunately , using az aks get-versions to list versions available for upgrade is kind of misleading and fixing it is better now.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/commands.py [hunk 1]
```diff
@@ -57,7 +58,7 @@ def load_command_table(self, _):
         g.custom_command('create', 'aks_create', no_wait_param='no_wait')
         g.command('delete', 'delete', no_wait_param='raw', confirmation=True)
         g.custom_command('get-credentials', 'aks_get_credentials')
-        g.command('get-versions', 'get_upgrade_profile', table_transformer=aks_get_versions_table_format)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/commands.py]: If you like to ease the impact of the breaking change, you can still keep the old entry, but use the `deprecate_info` to tell users to use the new one. In the help, add `Deprecated` to the short summary of the old one. After one release, you can pull the old stuff out. Up to you though.
```python
 g.command('get-versions', 'get_upgrade_profile', table_transformer=aks_get_versions_table_format, deprecate_info="For accuracy, 'get-versions' has been renamed to 'get-upgrades'. Please start to use the new one")
```

Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/commands.py]: That's great 👍, I wasn't aware of the deprecation feature. I'll add that later today.

**Comment by sauryadas** [comment]
The deprecation info is going to be super useful .. thanks both!!

Best
Saurya

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/commands.py (hunk 1 above)
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/commands.py]: @yugangw-msft I added a deprecation notice. Please let me know if there's anything else to change here. I'd really like to get this merged today for the next release (also #5466). 😄

**Comment by bureado** [comment]
Since I just hit this issue, you split `get-versions` and `get-upgrades` so that people can access basically the same information but before and after creating the cluster, correct?

If so, why does `get-versions` also returns upgrade paths?

**Comment by mboersma** [comment]
> why does `get-versions` also return upgrade paths?

Because that's considered useful associated information, but more specifically because the raw JSON response for both commands is basically the same. The real difference is `get-versions` returns versions and upgrades available *in general*, while `get-upgrades` returns only the upgrade versions available to a specific cluster.

**Pull request merged**